src/librbd/journal/Types.h

```diff
@@ -135,9 +136,10 @@ struct SnapCreateEvent : public SnapEventBase {
 
   SnapCreateEvent() {
   }
-  SnapCreateEvent(uint64_t op_tid, const std::string &snap_name)
-    : SnapEventBase(op_tid, snap_name) {
+  SnapCreateEvent(uint64_t op_tid, const std::string &snap_name, const cls::rbd::SnapshotNamespaceEntry &_snap_namespace)
```

Comment:
Would there be a need to journal group snapshot create events? It seems like the group actions would be journaled via a group journal (eventually). If group snap create events are recorded with the image, they wouldn't be properly associated with a group and you wouldn't be able to remove them (perhaps). Therefore, perhaps this should wait until after group actions can be journaled and replayed.